### PR TITLE
SensoApi can now take all query params from front end and query the senso /rate api

### PR DESCRIPTION
**Pull request by Shalev-Lifshitz**
The current SensoApi.queryApi method had a hard-coded string being passed to the /rate API. We could only set budget via method params.

**Description of changes**:
Created the SensoRateQuery class which is essentially a wrapper for all the params going to the /rate API. This gets passed to the Gson package which auto-creates a JSON string (which is passed to the /rate API). 

Also added more spring params to the /demo endpoint which are cast and passed to SensoApi.queryApi.

Removed JsonMap class as we are using Gson package instead.

## TODO next
The frontend now needs to be updated so that it sends all of the required params to the /demo endpoint.

### CODE REVIEW

**Comment by Shalev-Lifshitz** [comment]
Forgot to mention that this PR completes cards #21 and #23

File: src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package tech.autodirect.api.upstream;
+
+/**
+ * The SensoRateQuery class represents a single query to the SensoApi/rate api and contains
+ * all the information that will be passed as arguments when querying.
+ */
+public class SensoRateQuery {
```
Comment (sammdu) [inline src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java:7]: I think we can simply use a Map for this, but making a dedicated class is also okay

**Review by sammdu — DISMISSED** [review]

File: src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java (hunk 1 above)
Comment (sammdu) [reply src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java:7]: @Shalev-Lifshitz

File: src/main/java/tech/autodirect/api/ApiEndpoints.java [hunk 2]
```diff
@@ -44,9 +44,19 @@ public static void main(String[] args) {
 
 	// demo API endpoint; use for reference
 	@GetMapping("/demo")
-	public Object hello(@RequestParam String budget) {
+	public Object hello(@RequestParam String loanAmount, @RequestParam String creditScore, @RequestParam String budget,
+						@RequestParam String vehicleMake, @RequestParam String vehicleModel,
```
Comment (sammdu) [inline src/main/java/tech/autodirect/api/ApiEndpoints.java:48]: Can you just hard code the vehicle info for now?
It makes more sense to just supply the loanAmount, credit score, and budget.
Just edit this file here it's fine, you can keep the concrete class unchanged.

File: src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java (hunk 1 above)
Comment (Shalev-Lifshitz) [reply src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java:7]: Good point. The examples I looked at with Gson used a dedicated class, I'll try it with a HashMap. @sammdu

File: src/main/java/tech/autodirect/api/ApiEndpoints.java (hunk 2 above)
Comment (Shalev-Lifshitz) [reply src/main/java/tech/autodirect/api/ApiEndpoints.java:48]: Yeah, no problem. @sammdu

**Comment by sammdu** [comment]
@Shalev-Lifshitz I refactored you code a little bit, and fixed a couple bugs with the request. Read my changes a bit and lmk if you have questions.

Also I think #15 is also addressed.

**Review by sammdu — APPROVED** [review]

**Pull request merged**